### DISCUSSION TITLE
Rendering ketcher-react plugin with the initialize data

### DISCUSSION THREAD

**Discussion by mustafacanp** [epam/ketcher#586] [Q&A] [answered]
I need to initialize the `Editor` with the molecule data. How can I do it on the react side? I don't want to use the `setMolecule` with `window` object, because it will cause problems in the future.

I need to pass initialize data as a prop as like:
```
<Editor moleculeData={data} staticResourcesUrl={staticResourcesUrl} structServiceProvider={structServiceProvider} />
```

**Comment by mustafacanp** [comment]
Any thoughts, please? @AndreiMazol

**Reply by AndreiMazol** [reply]
Hi @mustafacanp ,

it will be implemented in  scope of release 2.3, issue https://github.com/epam/ketcher/issues/601

Best regards,
Andrei

**Reply by mustafacanp** [reply]
Thank @AndreiMazol.

And I want to initialize multiple editors on the same page. Is there any way to access the `Ketcher` object without the `window` object? I want to set and get different molecule data to different editors. e.g:

```
ketcher1.setMolecule(moleculeData1);
ketcher2.setMolecule(moleculeData2);
```

**Reply by mustafacanp** [reply]
What is the estimated release date of version 2.3, please?

**Reply by AndreiMazol** [reply]
We plan to add this functionality in scope of release 2.3 too.

**Reply by AndreiMazol** [reply]
Ketcher 2.3 should be released in 2 weeks.

**Reply by mustafacanp** [reply]
Okay thanks a lot

**Comment by AndreiMazol** [comment, marked as answer]
Hi @mustafacanp ,

Now you can get access to ketcher instance by passing an  additional property to Editor component. See example of usage below:

```
<Editor 
      staticResourcesUrl={process.env.PUBLIC_URL}
      structServiceProvider={structServiceProvider}
      onInit={ketcher => {
        setTimeout(() => {
          ketcher.setMolecule('CCCCCCCCCC>>CCCCCCCCC')
        }, 5000)
      }}
    />
```
